src/Http/Http.Extensions/gen/StaticRouteHandlerModel/StaticRouteHandlerModel.Emitter.cs

```diff
@@ -97,9 +96,31 @@ public static void EmitRequestHandler(this Endpoint endpoint, CodeWriter codeWri
         {
             codeWriter.WriteLine("return Task.CompletedTask;");
         }
+
         codeWriter.EndBlock(); // End handler method block
     }
 
+    private static void EmitHttpResponseContentType(this EndpointResponse endpointResponse, CodeWriter codeWriter)
+    {
+        if (!endpointResponse.HasNoResponse
+            && endpointResponse.ResponseType is { } responseType
+            && (responseType.SpecialType == SpecialType.System_Object || responseType.SpecialType == SpecialType.System_String))
+        {
+            codeWriter.WriteLine("if (result is string)");
+            codeWriter.StartBlock();
+            codeWriter.WriteLine($@"httpContext.Response.ContentType ??= ""text/plain; charset=utf-8"";");
+            codeWriter.EndBlock();
+            codeWriter.WriteLine("else");
+            codeWriter.StartBlock();
+            codeWriter.WriteLine($@"httpContext.Response.ContentType ??= ""application/json"";");
+            codeWriter.EndBlock();
+        }
+        else
+        {
+            codeWriter.WriteLine($@"httpContext.Response.ContentType ??= ""text/plain"";");
```

Comment:
Seems wrong? The default in RDF is json + charset. Which is applied via the `WriteAsJson` methods.

Reply:
Yeah, I think you are right. I actually found a few test cases what were incorrect. I'll check this against current RDF behavior and change as necessary.